sys/malloc_monitor/doc.txt

```diff
@@ -0,0 +1,71 @@
+/**
+@defgroup   sys_malloc_monitor Heap Memory Usage Monitor
+@ingroup    sys_memory_management
+@brief      This module allows to monitor the dynamic memory usage of a certain piece of code.
+@warning    This module automatically selects @ref sys_malloc_ts and naturally
+            incurs a certain runtime overhead. It is not meant for production usage.
+
+# Description
+
+This module allows to monitor the dynamic memory usage of a certain piece of code.
+It works by hooking into (wrappers to) @ref malloc(), @ref calloc(), @ref realloc(),
+and @ref free() calls to internally record the current and all-time maximum heap memory usage.
+
+Note that in general dynamic memory management is a bad idea on the constrained devices RIOT
+is targeting. So maybe it is better to just adapt your code to use static memory management instead.
+
+# Usage
+
+Enable the module with `USEMODULE += malloc_monitor`.
+
+Add `#include "malloc_monitor.h"` to the file in which you want to monitor dynamic memory usage.
+Use @ref malloc_monitor_get_usage_current() to retrieve the size of the currently allocated
+heap memory in bytes. @ref malloc_monitor_get_usage_high_watermark() returns the all-time maximum
+since startup or the last call to  @ref malloc_monitor_reset_high_watermark().
+
+Note that `malloc_monitor` currently has no notion of threads and will at any point in time report
+the global dynamic memory usage, not the one used by the currently running thread.
+Thread-safety is achieved through usage of @ref sys_malloc_ts, though.
+
+## Example
+
+Imagine you want to investigate the dynamic memory consumption of a certain function `func()`.
+The following snippet could get you started:
+
+```c
```

Comment:
```suggestion
``` C
```

Maybe this pleases Doxygen? (A wild guess only, though)

Reply:
Let's try!

Reply:
So [the fix](https://github.com/RIOT-OS/RIOT/pull/20363/commits/e54029f9ff12dc2f9906f92b16863993b35fb356) was to use the C documentation comment-style with explicit ` * ` at the beginning of each line.